pkg/upload/react-native-ios.go

```diff
@@ -2,45 +2,46 @@ package upload
 
 import (
 	"fmt"
+	"github.com/bugsnag/bugsnag-cli/pkg/ios"
+	"github.com/bugsnag/bugsnag-cli/pkg/server"
+	"github.com/bugsnag/bugsnag-cli/pkg/utils"
 	"os"
 	"path/filepath"
 
-	"github.com/bugsnag/bugsnag-cli/pkg/ios"
 	"github.com/bugsnag/bugsnag-cli/pkg/log"
 	"github.com/bugsnag/bugsnag-cli/pkg/options"
-	"github.com/bugsnag/bugsnag-cli/pkg/server"
-	"github.com/bugsnag/bugsnag-cli/pkg/utils"
 )
 
 func ProcessReactNativeIos(options options.CLI, endpoint string, logger log.Logger) error {
 	iosOptions := options.Upload.ReactNativeIos
-	var rootDirPath string
-	var plistData *ios.PlistData
-	var buildSettings *ios.XcodeBuildSettings
-	var err error
+	var (
+		rootDirPath string
+		plistData   *ios.PlistData
+		//buildSettings    *ios.XcodeBuildSettings
+		err              error
+		xcodeArchivePath string
+		buildDirPath     string
+	)
 
 	for _, path := range iosOptions.Path {
-		// Check/Set the build folder
-		buildDirPath := filepath.Join(path, "ios", "build")
-		rootDirPath = path
-		if !utils.FileExists(buildDirPath) {
-			buildDirPath = filepath.Join(path, "build")
-
-			if utils.FileExists(buildDirPath) {
-				rootDirPath = filepath.Join(path, "..")
-
-			} else if iosOptions.ReactNative.Bundle == "" || iosOptions.ReactNative.SourceMap == "" {
-				return fmt.Errorf("unable to find bundle files or source maps in within %s", path)
+		if filepath.Ext(path) == ".xcarchive" {
```

Comment:
I would really like to see this code extracted and shared with the dsym commands – something like "findLatestArchive" that takes the path and returns the archive path using this procedure. (I wonder if there's also an argument to say that we should search a directory for an archive file first, before looking for a project file.)

Then we can be sure that both bundles and dSYMs will follow the same logic.

Reply:
Discussed in IRL - it's probably not a good idea to search a directory for an archive as well as it just complicates the possibilities. And that means the code is as abstracted as it needs to be.